### PR TITLE
move libraries from libexec to lib

### PR DESCRIPTION
**Pull request by reidpr**
This PR addresses #596. That issue also proposes moving the test suite to the root of `libexec`. Even though `test` is the only entry remaining in `libexec`, I did not move the test suite up, because I anticipate growing some new executables in `libexec`.

I believe it also addresses #648. @wiene, can you comment? Should it also close that bug?

It may also address #683. @ana, what is your opinion on that? Would you like that issue to remain open due to the request for a build system variable, or be closed because it eliminates the need for your patch, or something else?

Thanks to @ana for providing a patch that forms part of this PR.

### CODE REVIEW

**Comment by ana** [comment]
Thanks @reidpr, this solves the need for a patch in openSUSE. I still think the hard coded path is a bad idea as mentioned in https://github.com/hpc/charliecloud/issues/683#issuecomment-595780211 but you can close the issue if you wish.

File: .gitignore [hunk 1]
```diff
@@ -11,11 +11,11 @@ a.out
...
-/libexec/charliecloud
-/libexec/contributors.bash
-/libexec/version.py
-/libexec/version.sh
-/libexec/version.txt
+/lib/charliecloud
+/lib/contributors.bash
+/lib/version.py
+/lib/version.sh
+/lib/version.txt
```
Comment (j-ogas) [inline .gitignore]: I'm not sure I like putting these generically named files directly into`/lib`. What are your thoughts about putting these files in a charliecloud sub directory, for example`/lib/charliecloud`, as the default behavior?

File: test/travis.yml [hunk 2]
```diff
@@ -28,10 +28,10 @@ compiler: gcc
 # weird errors, e.g.:
 #
 #   make[2]: Entering directory [...]
-#    /bin/mkdir -p '/var/tmp/libexec/charliecloud'
-#    ../yes base.sh '/var/tmp/libexec/charliecloud'
+#    /bin/mkdir -p '/var/tmp/lib/charliecloud'
+#    ../yes base.sh '/var/tmp/lib/charliecloud'
```
Comment (j-ogas) [inline test/travis.yml]: It looks like this path is expecting a charliecloud parent directory, e.g., `/lib/charliecloud`.

**Comment by wiene** [comment]
> I believe it also addresses #648. @wiene, can you comment?

Why do you think it does?

**Comment by reidpr** [comment]
@wiene, the files you flagged as being executable but without a shebang should now be not executable (and still without a shebang).

**Comment by wiene** [comment]
I've just tried out this PR and the mentioned files are actually not executable anymore. Thus I think that you are right that this patch fixes #648.

**Pull request merged**